### PR TITLE
Generate one-time Homebrew tarball

### PR DESCRIPTION
**Pull request by lissyx**

### CODE REVIEW

File: taskcluster/homebrew-build.sh [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  export HOMEBREW_LOGS="${_brew_instance}/homebrew.logs/"
+  export HOMEBREW_CACHE="${_brew_instance}/homebrew.cache/"
+
+  # Never fail on pre-existing homebrew/ directory
+  mkdir -p "${_brew_instance}" || true
+  mkdir -p "${HOMEBREW_CACHE}" || true
+
+  # Make sure to verify there is a 'brew' binary there, otherwise install things.
+  if [ ! -x "${_brew_instance}/bin/brew" ]; then
+    curl -L ${BREW_URL} | tar xz --strip 1 -C "${_brew_instance}"
```
Comment (reuben) [inline taskcluster/homebrew-build.sh]: Maybe we should use this opportunity to also pin the formulas? That way we avoid breakage when the formula language changes in a backwards incompatible way.

Comment (reuben) [reply taskcluster/homebrew-build.sh]: For example by saving a `${BREW_FORMULAS_COMMIT}` and then checking out that commit in `` `brew --prefix`/Homebrew/Library/Taps/homebrew/homebrew-core``

File: taskcluster/homebrew-build.sh [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      nvm install ${node_ver}
+    done;
+
+  install_pkg_homebrew "openssl"
+  install_pkg_homebrew "pkg-config"
+  install_pkg_homebrew "readline"
+  install_pkg_homebrew "sox"
+}
+
+if [ "${flavor}" = "--local" ]; then
```
Comment (reuben) [inline taskcluster/homebrew-build.sh]: My understanding is that "--local" is used for builds and "--tests" for tests. So maybe call it "--builds" and "--tests"?

File: taskcluster/homebrew-build.sh (hunk 1 above)
Comment (lissyx) [reply taskcluster/homebrew-build.sh]: > Maybe we should use this opportunity to also pin the formulas? That way we avoid breakage when the formula language changes in a backwards incompatible way.

Last that I checked, this was explicitely not supported. Aren't we safe with version-URL + the no auto update env var ?

Comment (reuben) [reply taskcluster/homebrew-build.sh]: The problem is that when you install it and run it, it clones the master branch of the formulas repo, regardless of the version of Homebrew itself.

Comment (lissyx) [reply taskcluster/homebrew-build.sh]: > The problem is that when you install it and run it, it clones the master branch of the formulas repo, regardless of the version of Homebrew itself.

Honestly, homebrew is a mess.

**Pull request merged**